File: icpc/2025_hochiminh_regional/l.py
import sys

def interact(i) :
    print(i)
    sys.stdout.flush()
    re = input()
    return re == "Y"

t = int(input())
while (t > 0) :
    c = 0
    while (1) :
        re = interact(c%2)
        if (re) :
            if (c != re) : interact(1)
            break
        c = 1-c
    t -= 1

    p = 1
    cnt = 1
    while (1) :
        re = interact(1)
        if (p == re) : break
        p = re
        cnt += 1

    print(-1, cnt)
    sys.stdout.flush()

File: icpc/2025_hochiminh_regional/l.cpp
#include <bits/stdc++.h>
using namespace std;

int qu=0;

int interact(int mark){
    qu++;
    cout << mark << '\n'; fflush(stdout);
    char c; cin >> c;
    return c=='Y';
}

void solve(){
    int c=0;
    while (1){
        int re = interact(c%2);
        if (re){
            if (c!=re){
                interact(0);
            }
            break;
        }
        c = 1-c;
    }
    int p=1, cnt=1;
    while (1){
        int re = interact(1);
        if (p == re) break;
        p = re;
        cnt ++;
    }
    cout << "-1 " << cnt << '\n';
    cout << " = " << qu << '\n';
    fflush(stdout);
}

int main(){
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}